crates/uv-resolver/src/lock/export/pylock_toml.rs

```diff
@@ -0,0 +1,565 @@
+use jiff::Timestamp;
+use toml_edit::{value, Array, ArrayOfTables, Item, Table};
+use url::Url;
+
+use uv_configuration::{DependencyGroupsWithDefaults, ExtrasSpecification, InstallOptions};
+use uv_distribution_types::{IndexUrl, RegistryBuiltWheel, RemoteSource, SourceDist};
+use uv_fs::{relative_to, PortablePathBuf};
+use uv_git_types::GitOid;
+use uv_normalize::{ExtraName, GroupName, PackageName};
+use uv_pep440::Version;
+use uv_pep508::MarkerTree;
+use uv_pypi_types::{Hashes, VcsKind};
+use uv_small_str::SmallString;
+
+use crate::lock::export::ExportableRequirements;
+use crate::lock::{each_element_on_its_line_array, LockErrorKind, Source};
+use crate::{Installable, LockError, RequiresPython};
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct PylockToml {
+    lock_version: Version,
+    created_by: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    requires_python: Option<RequiresPython>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    extras: Vec<ExtraName>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    dependency_groups: Vec<GroupName>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    default_groups: Vec<GroupName>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    packages: Vec<PylockTomlPackage>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    attestation_identities: Vec<PylockTomlAttestationIdentity>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct PylockTomlPackage {
+    name: PackageName,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    version: Option<Version>,
+    #[serde(
+        skip_serializing_if = "uv_pep508::marker::ser::is_empty",
+        serialize_with = "uv_pep508::marker::ser::serialize",
+        default
+    )]
+    marker: MarkerTree,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    requires_python: Option<RequiresPython>,
+    #[serde(skip_serializing_if = "Vec::is_empty", default)]
+    dependencies: Vec<PylockTomlDependency>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    index: Option<Url>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    vcs: Option<PylockTomlVcs>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    directory: Option<PylockTomlDirectory>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    archive: Option<PylockTomlArchive>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sdist: Option<PylockTomlSdist>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    wheels: Option<Vec<PylockTomlWheel>>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct PylockTomlDependency;
```

Comment:
This is intentionally a stub, right?

Reply:
Yeah. The PEP actually doesn't define a schema for this at all... You can put "whatever" you want in here.